server/lomas_server/dp_queries/dummy_dataset.py

```diff
@@ -38,102 +39,62 @@ def make_dummy_dataset(  # pylint: disable=too-many-locals
     Returns:
         pd.DataFrame: dummy dataframe based on metadata
     """
-    # Setting seed
-    random.seed(seed)
-    np.random.seed(seed)
+    # Creating new random generator with fixed seed
+    rng = np.random.default_rng(seed)
 
     # Create dataframe
     df = pd.DataFrame()
-    for col_name, data in metadata["columns"].items():
+    for col_name, data in metadata.columns.items():
         # Create a random serie based on the data type
-        match data["type"]:
-            case "string":
-                if "cardinality" in data.keys():
-                    cardinality = data["cardinality"]
-                    if "categories" in data.keys():
-                        categories = data["categories"]
-                        serie = pd.Series(
-                            random.choices(
-                                categories, k=nb_rows  # type: ignore
-                            )
-                        )
-                    else:
-                        serie = pd.Series(
-                            random.choices(
-                                RANDOM_STRINGS[:cardinality], k=nb_rows
-                            )
-                        )
-                else:
-                    serie = pd.Series(
-                        random.choices(RANDOM_STRINGS, k=nb_rows)
-                    )
-            case "boolean":
-                # type boolean instead of bool will allow null values
+        match data:
+            case CategoricalColumnMetadata():
+                categories = data.categories
+                serie = pd.Series(rng.choice(categories, size=nb_rows))
+            case StrMetadata():
+                serie = pd.Series(rng.choice(RANDOM_STRINGS, size=nb_rows))
```

Comment:
It should take the values within "categories" which must be there.
We can delete RANDOM_STRING constatnt also

Reply:
it is like CategoricalColumnMetadata